File: 子串/560.和为k的子数组.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        Map<Integer,Integer>map = new HashMap<>();
        int res=0;
        int sum=0;
        map.put(0,1);
        for(int x:nums){
            sum+=x;
            res+=map.getOrDefault(sum-k,0);
            map.put(sum,map.getOrDefault(sum,0)+1);
        }
        return res;

    }
}

File: 子串/560.和为k的子数组.go
package 子串

func subarraySum(nums []int, k int) int {
	cnt := make(map[int]int, len(nums))
	sum := 0
	ans := 0
	cnt[0] = 1
	for _, v := range nums {
		sum += v
		ans += cnt[sum-k]
		cnt[sum]++
	}
	return ans
}
